File: 55. Jump Game/55. Jump Game.java
class Solution {
    public boolean canJump(int[] nums) {
        int maxReach = 0;
        
        for (int i = 0; i <= maxReach && i < nums.length; i++) {
            maxReach = Math.max(maxReach, i + nums[i]);
            if (maxReach >= nums.length - 1) {
                return true;
            }
        }
        
        return false;
    }
}

File: 55. Jump Game/55. Jump Game.go
func canJump(nums []int) bool {
    maxReach := 0
    
    for i := 0; i <= maxReach && i < len(nums); i++ {
        if i + nums[i] > maxReach {
            maxReach = i + nums[i]
        }
        if maxReach >= len(nums) - 1 {
            return true
        }
    }
    
    return false
}